python/plugins/processing/tools/vector.py

```diff
@@ -521,6 +523,21 @@ def ogrLayerName(uri):
         regex = re.compile('(layername=)(.*)')
         r = regex.search(uri)
         return r.groups()[1]
+    elif '.gdb' in uri:
```

Comment:
why triggering on .gdb only and not layerid= ? Could happen with other multi-layer formats.